pandas/tests/plotting/frame/test_frame_color.py

```diff
@@ -204,7 +206,11 @@ def test_scatter_with_c_column_name_with_colors(self, cmap, kw):
             columns=["length", "width"],
         )
         df["species"] = ["r", "r", "g", "g", "b"]
-        ax = df.plot.scatter(x=0, y=1, cmap=cmap, **{kw: "species"})
+        if mpl_ge_3_6_0() and cmap is not None:
+            with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
```

Comment:
does this indicate that we are doing something wrong/outdated?

Reply:
@MarcoGorelli looks like you raised and addressed this issue in https://github.com/pandas-dev/pandas/pull/34344

Do you remember if passing `cmap` in `plot.scatter was necessary` to test this behavior?

Reply:
FWIW: I think this is a new warning in mpl 3.6 so `plot.scatter` may have been doing something "wrong" before?

Reply:
It was silently ignored before. They added stricter checks in a few places. If we want to check that the colormap is not applied, then we have to check for the user warning